File: gifts.py

# Read in number of visits
n = int(input())

# Loop over the visits
for i in range(n):
   # Read in the 3 gifts
   a, b, c = map(int, input().split())

   # Output the smallest difference
   print(min(abs(a-b), abs(a-c), abs(b-c)))

File: gifts.cpp
#include <stdio.h>
#include <stdlib.h>

static void one_case(void) {
   int a, b, c;

   // Read in the three gifts
   scanf("%d %d %d", &a, &b, &c);

   // Default min difference
   int diff = abs(b - a);

   // Now check all other pairings
   if (abs(c - a) < diff)
      diff = abs(c - a);
   if (abs(c - b) < diff)
      diff = abs(c - b);
  
   // Print result
   printf("%d\n", diff);
}

int main(void) {
   // Read the number of gifts
   int t;
   scanf("%d", &t);

   // Loop over the cases
   while(t--) one_case();

   return 0;
}

